src/plan/barriers.rs

```diff
@@ -48,8 +52,16 @@ impl<E: ProcessEdgesWork> ObjectRememberingBarrier<E> {
     }
 
     #[inline(always)]
-    fn enqueue_node(&mut self, obj: ObjectReference) {
-        if compare_exchange_atomic(self.meta, obj.to_address(), 0b1, 0b0) {
+    fn enqueue_node<VM: VMBinding>(&mut self, obj: ObjectReference) {
+        if VM::VMObjectModel::compare_exchange_metadata(
+            self.meta,
```

Comment:
`self.meta` is probably not a constant. This means you may not be able to actually use const generics for those VMObjectModel methods.

Reply:
It is a constant, but Rust may not see it as a constant when it is a member of an object.
Anyway, I don't think there is any reason that we can not directly use the `VM::VMObjectModel::GLOBAL_LOG_BIT_SPEC` constant here

Reply:
We still need `.meta` to implement the `Barrier` trait, but for the `ObjectRememberingBarrier` struct methods such as `enqueue_node`, we always refer to a constant.
@wenyuzhao What do you think? Is there any reason not to directly use `VM::VMObjectModel::GLOBAL_LOG_BIT_SPEC` here?

Reply:
Theoretically it should be possible to use const generics for our use case.
However, the feature seems very unstable/immature. It doesn't show useful compiler errors and there is not much online information about how to solve the issues I face.
I think we should defer this to some time in the future when the feature is more stable.